### PR TITLE
TextMaskedInput typing improvements

### PR DESCRIPTION
**Pull request by roblevintennis**
**Description:**
- [Asana Task](https://app.asana.com/0/1137984670709817/1155343635039372/f)- 
- TextMaskedInput typings

**Verification**
`$ yarn test:types` with no errors

### CODE REVIEW

File: src/components/index.d.ts [hunk 1]
```diff
@@ -373,40 +373,40 @@ export declare const EmailInput: {
...
+    type: string
+    'data-tid': string
+    name: string
+    labelCopy: string
+    doValidation?: any
+    placeholder?: any
+    guide?: boolean
+    initialValue?: string
+    keepCharPositions?: boolean
+    pipe?: any
```
Comment (bwalker8038) [inline src/components/index.d.ts]: If this is a function, the type signature should be something like this:

```ts
  pipe?: () => void // or any other return type: any, T (Generic), boolean, etc)
```

File: src/components/index.d.ts [hunk 2]
```diff
@@ -373,40 +373,40 @@ export declare const EmailInput: {
...
+    disabled?: boolean
+    allCaps?: boolean
+    validator?: any
```
Comment (bwalker8038) [inline src/components/index.d.ts]: see above

File: src/components/index.d.ts [hunk 3]
```diff
@@ -373,40 +373,40 @@ export declare const EmailInput: {
...
-    labelCopy: any
-    validator: any
-    setTouched: any
-    getTouched: any
+    type: string
+    'data-tid': string
+    name: string
+    labelCopy: string
+    doValidation?: any
+    placeholder?: any
```
Comment (bwalker8038) [inline src/components/index.d.ts]: Nit: this should be a `string`

File: src/components/index.d.ts [hunk 4]
```diff
@@ -373,40 +373,40 @@ export declare const EmailInput: {
...
-    name: any
-    labelCopy: any
-    validator: any
-    setTouched: any
-    getTouched: any
+    type: string
+    'data-tid': string
+    name: string
+    labelCopy: string
+    doValidation?: any
```
Comment (bwalker8038) [inline src/components/index.d.ts]: If this is a function, the type signature should be something like this:

```ts
  pipe?: () => void // or any other return type: any, T (Generic), boolean, etc)
```

Comment (roblevintennis) [reply src/components/index.d.ts]: Thanks @bwalker8038 -- I was going to save the functions for another pass, but I think I'll just go ahead and start on a follow up now!

**Pull request merged**